sormas-backend/src/main/java/de/symeda/sormas/backend/labmessage/LabMessageFacadeEjb.java

```diff
@@ -382,6 +387,27 @@ public boolean exists(String uuid) {
 		return labMessageService.exists(uuid);
 	}
 
+	@Override
+	public boolean existsLabMessageForEntity(ReferenceDto entityRef) {
+		List<Sample> samples;
+		if (CaseReferenceDto.class.equals(entityRef.getClass())) {
+			samples = sampleService.getByCaseUuids(Arrays.asList(entityRef.getUuid()));
```

Comment:
I was thinking to have a check directly in the DB and not retrieve all the samples for a case. I think in the past there were performance problems because of the number of samples.

I tried with a new method which I added in `LabMessageService` and the tests passed.

`LabMessageSerice.java`
```java
public long countForCase(String caseUuid) {
		CriteriaBuilder cb = em.getCriteriaBuilder();
		CriteriaQuery<Long> cq = cb.createQuery(Long.class);
		Root<LabMessage> labMessageRoot = cq.from(LabMessage.class);
		Join<LabMessage, Sample> sampleJoin = labMessageRoot.join(LabMessage.SAMPLE, JoinType.LEFT);
		Join<Sample, Case> caseJoin = sampleJoin.join(Sample.ASSOCIATED_CASE, JoinType.LEFT);

		Predicate filter = cb.and(createDefaultFilter(cb, labMessageRoot), caseJoin.get(AbstractDomainObject.UUID).in(Collections.singleton(caseUuid)));

		cq.where(filter);
		cq.select(cb.countDistinct(labMessageRoot));

		return em.createQuery(cq).getSingleResult();
	}
```

Then I changed this line to this
```suggestion
	return labMessageService.countForCase(entityRef.getUuid()) > 0;
```

The `countForCase` method can be replicated for Contact and EventParticipant